molecule_libvirt/playbooks/create.yml

```diff
@@ -0,0 +1,259 @@
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host if item.libvirt_host is defined else 'localhost' }}"
+
+    - name: "Create network {{ molecule_network_name }}"
+      virt_net:
+        command: define
+        name: "{{ molecule_network_name }}"
+        xml: "{{ lookup('template', 'network/network.xml.j2') }}"
+        uri: "{{ qemu_uri }}"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      when: item.network_name is not defined
+      vars:
+        qemu_uri: "{{ 'qemu+ssh://'+item.libvirt_user+'@'+item.libvirt_host+'/system' if item.libvirt_host is defined else libvirt_uri }}"
+
+    - name: "Start network {{ molecule_network_name }}"
+      virt_net:
+        command: start
+        name: "{{ molecule_network_name }}"
+        uri: "{{ 'qemu+ssh://'+item.libvirt_user+'@'+item.libvirt_host+'/system' if item.libvirt_host is defined else libvirt_uri }}"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      when: item.network_name is not defined
+
+    - name: Get qcow2 image
+      get_url:
+        url: "{{ image_url | default(default_image_url) }}"
+        dest: "{{ libvirt_default_image_path }}/{{ item.name }}.qcow2"
+        mode: '0660'
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Generate OpenSSH key pair
+      openssh_keypair:
+        path: "{{ molecule_private_key_path }}"
+        size: 2048
+      register: ssh_key
+      delegate_to: "localhost"
+
+    - name: Create temporary folder
+      file:
+        path: "{{ temp_folder }}/{{ item.name }}-molecule"
+        state: directory
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Create user-data from template
+      template:
+        src: user-data.j2
+        dest: "{{ temp_folder }}/{{ item.name }}-molecule/user-data"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Create meta-data from template
+      template:
+        src: meta-data.j2
+        dest: "{{ temp_folder }}/{{ item.name }}-molecule/meta-data"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Create cloud-init iso file
+      command:
+        "mkisofs -o
+        {{ libvirt_default_image_path }}/{{ item.name }}-cloud-init.iso
+        -V cidata -J -r user-data meta-data"
+      args:
+        chdir: "{{ temp_folder }}/{{ item.name }}-molecule"
+      with_items: "{{ molecule_yml.platforms }}"
```

Comment:
Do we want to keep `with_items` or do we want to use `loop`?

Reply:
with_items has been changed for loop.